daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/Error.scala

```diff
@@ -100,14 +100,14 @@ object Error {
         override val message: String,
     ) extends Error
 
-    final case class ValueNesting(value: Value[Value.ContractId]) extends Error {
+    final case class ValueNesting(culprit: Value[Value.ContractId]) extends Error {
       override def message: String =
         s"Provided value exceeds maximum nesting level of ${Value.MAXIMUM_NESTING}"
     }
 
-    final case class NonSuffixedCid(cid: Value.ContractId.V1) extends Error {
+    final case class IllegalContractId(cid: Value.ContractId) extends Error {
```

Comment:
should we include the reason here and tell users that it failed because of lack of suffix or because they used v0?

Reply:
I did not feel it was necessary, in the sens that if it's happen it means the user try to feed to the ledger ID that it has cooked himself. If the user want to shoot himself in the foot, I do not see why I shoot tell him which foot is it. 
If you really think it is useful, I will add the reason.

Reply:
fair enough, I tend to lean on the side of including too much information rather than too little in case we need it later but I don’t feel all that strongly about this one.

Reply:
88c5c2dabe5be062dab899680144636703e9f079